tests/integration/l1_handler_txn_tests.rs

```diff
@@ -0,0 +1,92 @@
+use std::sync::Arc;
+
+use blockifier::abi::abi_utils::selector_from_name;
+use blockifier::context::BlockContext;
+use blockifier::test_utils::{NonceManager, BALANCE};
+use blockifier::transaction::test_utils::{block_context, max_fee};
+use blockifier::transaction::transactions::L1HandlerTransaction;
+use futures::Future;
+use rstest::{fixture, rstest};
+use starknet_api::core::{ContractAddress, EntryPointSelector};
+use starknet_api::hash::StarkFelt;
+use starknet_api::stark_felt;
+use starknet_api::transaction::{Calldata, Fee, TransactionVersion};
+
+use crate::common::state::{
+    init_logging, initial_state_cairo0, load_cairo1_contract, StarknetStateBuilder, StarknetTestState,
+};
+use crate::common::transaction_utils::execute_txs_and_run_os;
+
+#[fixture]
+pub async fn l1_initial_state_cairo1(
```

Comment:
I don't think you need custom fixtures for this test, you can reuse the `cairo0` and `cairo1` fixtures used in most of the other tests.